internal/toolexec/processors/aspect/linkdeps/linkdeps.go

```diff
@@ -0,0 +1,146 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2023-present Datadog, Inc.
+
+package linkdeps
+
+import (
+	"bufio"
+	"fmt"
+	"io"
+	"os"
+	"sort"
+	"strings"
+)
+
+const (
+	// LinkDepsFilename is the standard file name for link.deps files.
+	LinkDepsFilename = "link.deps"
+
+	headerV1 = "#" + LinkDepsFilename + "@v1"
+)
+
+// LinkDeps represents the contents of a link.deps file.
+type LinkDeps struct {
+	deps map[string]struct{}
+}
+
+// ReadFile reads a link.deps file from the provided filename.
+func ReadFile(filename string) (LinkDeps, error) {
+	file, err := os.Open(filename)
+	if err != nil {
+		return LinkDeps{}, err
+	}
+	defer file.Close()
+
+	return Read(file)
+}
+
+// Read reads a link.deps file content from the provided reader.
+func Read(r io.Reader) (l LinkDeps, err error) {
+	rd := bufio.NewReader(r)
+
+	var line string
+	if line, err = rd.ReadString('\n'); err != nil {
+		return
+	}
+
+	switch hdr := strings.TrimSpace(line); hdr {
+	case headerV1:
+		return parseV1(rd)
+	default:
+		err = fmt.Errorf("unsupported data format %q, a newer Orchestion release may be required", hdr)
+		return
+	}
+}
+
+// parseV1 parses the contents of V1 link.deps files.
```

Comment:
What's the versioning for?

Reply:
This is somewhat future-proofing; so that if we want to change the content/format of this, we have an easy & early detection mechanism which affords us outputting an appropriate error message...